rfcs/0146-meta-categories.md

```diff
@@ -0,0 +1,204 @@
+   put the files on the filesystem hierarchy, deviating them from the job of
+   really writing them.
+
+   - Or just taking the shortest path and throw it on a folder under `misc`.
+
+6. It "locks" the filesystem, preventing its usage for other, more sensible
+   purposes.
+
+7. The most important: the categorization is not discoverable via Nix language
+   infrastructure.
+
+   Indeed there is no higher level way to query about such categories besides
+   the one described in the bullet 2 above.
+
+In light of such a bunch of problems, this RFC proposes a novel alternative to
+the above mess: new `meta` attributes.
+
+# Detailed design
+[design]: #detailed-design
+
+A new attribute, `meta.categories`, will be included for every Nix expression
+living inside Nixpkgs.
+
+This attribute will be a list, whose elements are one of the possible elements
+of the `lib.categories` set.
+
+A typical snippet of `lib.categories` will be similar to:
+
+```nix
+{
+  assembler = {
+    name = "Assembler";
+    description = ''
+      A program that converts text written in assembly language to binary code.
+    '';
+  };
+
+  compiler = {
+    name = "Compiler";
+    description = ''
+      A program that converts a source from a language to another, usually from
+      a higher, human-readable level to a lower, machine level.
+    '';
+  };
+
+  font = {
+    name = "Font";
+    description = ''
+      A set of files that defines a set of graphically-related glyphs.
+    '';
+  };
+
+  game = {
+    name = "Game";
+    description = ''
+      A program developed with entertainment in mind.
+    '';
+  };
+
+  interpreter = {
+    name = "Interpreter";
+    description = ''
+      A program that directly executes instructions written in a programming
+      language, without requiring compilation into the native machine language.
+    '';
+  };
+
+```
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+In file bochs/default.nix:
+
+```nix
+stdenv.mkDerivation {
+
+. . .
+
+  meta = {
+    . . .
+    categories = with lib.categories; [ emulator debugger ];
+    . . .
+    };
+  };
+}
+
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The most immediate drawbacks are:
+
+1. A huge treewide edit of Nixpkgs
+
+   On the other hand, this is easily sprintable and amenable to automation.
+   
+2. Bikeshedding
+
+   How many and which categories we should create? Can we expand them later? 
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Do nothing
+
+2. Ignore/nuke the categorization completely
+
+   This is not an idea as bad as it appear. After all, categorization has a
+   non-negligible propensity to bikeshedding. Removing it removes all problems.
+   
+   Nonetheless, other good software collections do this just fine, and we can
+   easily imitate them. Indeed, we can follow/take a peek at how Repology keeps
+   the categorizations defined by those software collections.
+
```

Comment:
Another alternative might be an open taxonomy.

Reply:
Well, I believe I can produce some preliminar code on this regard.
Basically I can verbatim copy-paste the categories of Freedesktop, just for start.

Reply:
Here's a largely uncontrolled categorization, although skewed by a built in scaffolding tool that only suggests some 20 categories: [Haskell packages by category](https://hackage.haskell.org/packages/)

My takeaways:

 - listing an uncurated set of categories does not appear to be useful
    - curating and restricting are different ideas!
 - some categories highlight library ecosystems (e.g. the Scotty category) - this could alternatively be captured by a related packages attribute (presumably out of scope, but worth considering)
 - we might want to embed these categories into ours - if we go with some open categories

Reply:
I have proposed something like "related categories" on the preliminary code:

https://github.com/NixOS/nixpkgs/pull/230439